src/main/scala/uk/gov/nationalarchives/DiscoveryService.scala

```diff
@@ -20,10 +20,16 @@ import scala.xml.XML
 class DiscoveryService(discoveryBaseUrl: String, backend: SttpBackend[IO, Fs2Streams[IO]], randomUuidGenerator: () => UUID) {
   private val folder = Folder
 
+  private def replaceUnicodeDecimals(input: String) =
+    "&#[0-9]+".r.replaceAllIn(input, _.matched.drop(2).toInt.toChar.toString)
+
   private def stripHtmlFromDiscoveryResponse(discoveryAsset: DiscoveryCollectionAsset) = {
     val resources = for {
       xsltStream <- Resource.make(IO(getClass.getResourceAsStream("/transform.xsl")))(is => IO(is.close()))
-      inputStream <- Resource.make(IO(new ByteArrayInputStream(discoveryAsset.scopeContent.description.getBytes())))(is => IO(is.close()))
+      inputStream <- Resource.make {
+        val description = replaceUnicodeDecimals(discoveryAsset.scopeContent.description)
```

Comment:
Maybe `descriptionWithHtmlCodesReplaced` or `descriptionWithDecimalsCharsReplaced`?

Reply:
`descriptionWithHtmlCodesReplaced` is Ok, I've changed it.